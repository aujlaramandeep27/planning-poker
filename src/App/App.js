import {Route} from "react-router-dom";
import React from 'react';
import Host from '../Host/Host'
import Meetings from '../Meetings/Components/Meetings/Meetings';
import Estimate from '../Estimate/Components/Estimate';
import CreateMeeting from '../Meetings/Components/CreateMeeting/CreateMeeting';
import CreateUser from '../User/Components/CreateUser';
import AddTicket from '../Tickets/AddTicket/AddTicket';
import Story from '../Story/Story';
import {history} from "../store";
import {ConnectedRouter} from "connected-react-router";

export default function App() {

  return (
      <ConnectedRouter history={history}>
          <Route exact path="/" component={CreateUser}/>
          <Route path="/host" component={Host} />
          <Route path="/meetings/" component={Meetings} />
          <Route path="/meeting/create" component={CreateMeeting} />
          <Route path="/estimate/" component={Estimate} />
          <Route path="/tickets/add" component={AddTicket} />
          <Route path="/story/:storyId" component={Story} />
      </ConnectedRouter>
  );
}
