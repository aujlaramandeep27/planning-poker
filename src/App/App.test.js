import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(
    {
        user: {
            items: null,
            error: null,
            loading: false
        }
    });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  expect(App()).toMatchSnapshot(); 
});
