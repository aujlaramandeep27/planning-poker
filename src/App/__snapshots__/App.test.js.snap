// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Memo(ConnectFunction)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CreateUser)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Stories)",
        "type": [Function],
      }
    }
    path="/stories/"
  />
  <Route
    component={[Function]}
    path="/meetings/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CreateMeeting)",
        "type": [Function],
      }
    }
    path="/meeting/create/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(StoryEstimate)",
        "type": [Function],
      }
    }
    path="/estimate/:meetingId/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CreateStory)",
        "type": [Function],
      }
    }
    path="/story/create/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(StorySummary)",
        "type": [Function],
      }
    }
    path="/story/summary/:storyId/"
  />
</Memo(ConnectFunction)>
`;
